src/features/Lock/lock.js

```diff
@@ -55,7 +55,7 @@ const Lock = () => {
   const [lockValue, setLockValue] = useState(7);
   const [currentHiIQ, setCurrentHiIQ] = useState(undefined);
   const [filledAmount, setFilledAmount] = useState();
-  const [openWrongChainModal, setOpenWrongChainModal] = useState(false);
+  const { setOpenWrongChainModal } = useContext(WrongChainContext);
```

Comment:
is there any reason why the effect that checks the wallet to show wrong chain modal can't be extracted to the layout component and removed from all the child ones?

Reply:
@kesar lemme see what can I do ✌